Simple-Projects/Simple_Projects/Gun/Gun.cpp

```diff
@@ -29,28 +30,30 @@ int Gun::getTotalShots() const {
 void Gun::prepare() {
     this->isReady = true;
 }
+
 void Gun::reload() {
     amount = capacity;
 }
+
 void Gun::shoot() {
     if (amount == 0) {
         throw OutOfRounds();
     }
     if (!isReady) {
         throw NotReady();
     }
-    amount -= 1;
-    totalShots += 1;
+    --amount;
+    ++totalShots;
     std::cout << "Bang!" << std::endl;
```

Comment:
The operator<< uses any supplied ostream, but this one writes directly to cout. Funny, but strange